lib/kennel/filter.rb

```diff
@@ -18,15 +16,27 @@ def filter_parts(parts)
       filter_resources(parts, :tracking_id, tracking_id_filter, "resources", "TRACKING_ID")
     end
 
+    def filtering?
+      !project_filter.nil?
+    end
+
+    def matches_project_id?(project_id)
+      return true unless filtering?
+
+      project_filter.include?(project_id)
```

Comment:
easier to follow + shorter ?
```suggestion
      filtering? && project_filter.include?(project_id)
```